### ISSUE TITLE
Behaviour of `primFloatEquality 0.0 -0.0`

### ISSUE THREAD

**Issue by asr** [agda/agda#2178]
After #2169 I noticed that `primFloatEquality 0.0 -0.0` is equal to `true`:

```
open import Common.Equality
open import Common.Prelude

wrong? : primFloatEquality 0.0 -0.0 ≡ true
wrong? = refl
```

Is it the expected behaviour? If not, it can be fixed using the bitwise equality (see https://github.com/idris-lang/Idris-dev/pull/3101).

**Comment by UlfNorell** [comment]
Duplicate of #2169. `primFloatEquality` needs to match definitional equality.

**Comment by asr** [comment]
Since the equalities in both issues are different, I don't understand why the issues are duplicate. In fact, I'll push a commit fixing this issue, but this commit doesn't fix #2169.

**Comment by UlfNorell** [comment]
Please don't. The equalities should match, so just changing `primFloatEquality` without fixing definitional equality makes the problem worse.

**Comment by asr** [comment]
OK. I can fix `primFloatEquality` replacing in `Agda.TypeChecking.Primitive`

``` haskell
floatEq :: Double -> Double -> Bool
floatEq x y | isNaN x && isNaN y = True
            | otherwise          = x == y
```

by (I have no tested the behaviour of `identicalIEEE` with NaN)

``` haskell
floatEq :: Double -> Double -> Bool
floatEq x y | isNaN x && isNaN y = True
            | otherwise          = identicalIEEE x y
```

Where I need to look for fixing the definitional equality?
